src/drt/src/gc/FlexGC_main.cpp

```diff
@@ -905,6 +911,192 @@ void FlexGCWorker::Impl::checkMetalSpacing()
+      if (prlLength < 0) {
+        logger_->warn(DRT,
+                      334,
+                      "Unsupported negative value for prlLength in branch PRL "
+                      "in checkMetalSpacing_wrongDir.");
+      }
+    }
+    auto spcVal = con->getWrongDirSpace();
+    // Loop over all edges of pin
+    for (auto& edges : pin->getPolygonEdges()) {
+      for (auto& edge : edges) {
+        // Check wrongDir edge
+        if (edge->isVertical() != layer->isVertical()) {
+          // Check noneol flag
+          if (con->hasNoneolWidth()) {
+            // Get edge length and compare
+            auto edgeLength = gtl::length(*edge);
+            auto noneolLength = con->getNoneolWidth();
+            if (edgeLength < noneolLength) {
+              continue;
+            }
+          }
+          gtl::rectangle_data<frCoord> rect1(edge->getLowCorner()->x(),
+                                             edge->getLowCorner()->y(),
+                                             edge->getHighCorner()->x(),
+                                             edge->getHighCorner()->y());
+          box_t queryBox;
+          checkMetalSpacing_wrongDir_getQueryBox(edge.get(), spcVal, queryBox);
+          vector<pair<segment_t, gcSegment*>> results;
+          auto& workerRegionQuery = getWorkerRegionQuery();
+          workerRegionQuery.queryPolygonEdge(queryBox, layerNum, results);
+          for (auto& [boostSeg, ptr] : results) {
+            // Check query edge wrongDir
+            if (ptr->isVertical() != layer->isVertical()) {
+              if (edge.get() == ptr) {
+                continue;
+              }
+
+              auto net1 = edge->getNet();
+              auto net2 = ptr->getNet();
+              if (net1 == net2) {
+                continue;
+              }
+
+              // no violation if fixed shapes
+              if (edge->isFixed() && ptr->isFixed()) {
+                continue;
+              }
+
+              // Get edges prl
+              const frDirEnum dir = edge->getDir();
+              const gtl::orientation_2d orient
+                  = (dir == frDirEnum::W || dir == frDirEnum::E)
+                        ? gtl::HORIZONTAL
+                        : gtl::VERTICAL;
+              const frCoord edge1_low = edge->low().get(orient);
+              const frCoord edge1_high = edge->high().get(orient);
+              const frCoord edge1_min = std::min(edge1_low, edge1_high);
+              const frCoord edge1_max = std::max(edge1_low, edge1_high);
+
+              const frCoord edge2_low = ptr->low().get(orient);
+              const frCoord edge2_high = ptr->high().get(orient);
+              const frCoord edge2_min = std::min(edge2_low, edge2_high);
+              const frCoord edge2_max = std::max(edge2_low, edge2_high);
+              const frCoord prl = std::min(edge1_max, edge2_max)
+                                  - std::max(edge1_min, edge2_min);
+              // Check prl run length > 0
```

Comment:
I feel this part is a replica of the prl check in checkMetalEndOfLine_eol_hasEol_check, Could you please reduce it to a callable method by both functions. In general we want to minimize code.